packages/core-state/src/block-state.ts

```diff
@@ -9,22 +9,30 @@ export class BlockState {
     private readonly app!: Contracts.Kernel.Application;
 
     @Container.inject(Container.Identifiers.WalletRepository)
+    @Container.tagged("state", "blockchain") // TODO: Without this line - and despite
```

Comment:
@rainydio Is this class supposed to use this tagged version? Or is it designed to work with different injected tags? If so - is there a way to make it explicit which tagged version a class is supposed to be used with?

Without this line, the test suite fails with Ambiguous match found for serviceIdentifier Repository<Wallet>. The Sandbox setup is initialised in more or less the same way as the ServiceProvider - I'm wondering if this would not fail also?

Reply:
It's supposed to be tag agnostic. It doesn't care which particular wallet repository it's using. You probably want to use getTagged if you want blockchain version. But definitely do not add that tag there.